### DISCUSSION TITLE
Hooked syscalls not triggering

### DISCUSSION THREAD

**Discussion by x1mus** [yanivagman/BPFroid#20] [Q&A]
Hello,

When performing the tracing of `java.io.File.exists()`, the output stops at the libc functions even if I decided to hook system calls. A system call such as `stat` or something else would actually be triggered but is not represented in the output trace (event if I don't filter on any package and use all possible syscalls).

By playing around with `strace` on a toy java example I figured out that the checking of the file is handled by a subprocess. (I tried using the --trace follow filter but no solution on this side as well).

**Comment by yanivagman** [comment]
What is the command line you use to trigger bpfroid (tracee)?

**Reply by x1mus** [reply]
I tried:
`./tracee --trace package="package" --trace follow --output json --output out-file:out.json`

And also variants where I don't filter on package and perform some keyword searching by hand.
`./tracee --trace follow --output json --output out-file:out.json`
`./tracee --output json --output out-file:out.json`

**Comment by yanivagman** [comment]
Try to provide the syscall name in the command line directly:
`--trace e=open,openat`
To trace all syscalls:
`--trace set=syscalls `

**Reply by x1mus** [reply]
Tried the proposed arguments with all variants specified above. No difference in the output.
Note: I just realized that when `java.io.File.exists()` is called from within the library itself (for example by StateService) the syscall is actually hooked it seems. But if the call is made from the user like in my case it doesnt trigger. This doesn't make sense to me.

**Reply by yanivagman** [reply]
Sorry, I have no idea. Doesn't make sense to me as well.
